gemrb/plugins/SDLVideo/OpenGLEnv.h

```diff
@@ -21,10 +21,16 @@
 #ifndef GemRB_OpenGLEnv_h
 #define GemRB_OpenGLEnv_h
 
-#if __APPLE__
+#define GL_GLEXT_PROTOTYPES 1
+
+#ifdef __APPLE__
 	#include <OpenGL/gl3.h>
 #else
-	#include <GL/glew.h>
+#ifdef USE_OPENGL_API
+	#include <SDL_opengl.h>
```

Comment:
That does not work for everything beyond OpenGL 1. While the header declares everything, there is nothing to link the OpenGL 2.0+ stuff against (like ` undefined reference to __imp_glCompileShader`). And for everything SDL2 may need internally, it does the procedure lookup [manually](https://github.com/spurious/SDL-mirror/blob/579d3b8ef46f995832a10561f2de743051874f36/src/render/opengl/SDL_shaders_gl.c#L473). So GLEW again makes that easy.

Reply:
This does work for OpenGL2 - as reported in #932 - though it might not work for OpenGL >= 3.
 SDL2 loads the GL implementation dynamically, if the GL library has the necessary support, it will not be a problem.

Regarding the error you get - is this from a clean compile ? I think I've seen this error during testing after removing GLEW, but with previous object files containing GLEW references

Reply:
I am not sure where in #932 you can imply this? Anyway, I really did a fresh build env to double-check. No change. Look at this link chain: ` -lSDL2 -lopengl32 `. None of them supports OpenGL2.0+ here by the required symbols. GLEW _does_ and links or sets up dispatches to their GL proc in some way when calling `glewInit()` at runtime. 

Thinking about it: you need to have all the `glX` stuff assigned by their runtime lookup results, and better only once. Therefore, you either do this yourself (using `SDL_GL_GetProcAddress `, or whatever your OS asks you), or you have it globally assigned by some compilation unit that has reserved space to be linked. That is what GLEW or epoxy are doing for the whole OpenGL2.0+ stuff, by eager or lazy loading. Apple prebakes this up to OGL3, so there is no need for GLEW either. ES seems to follow that but I cannot verify.

Look at this [minimal example](https://gist.github.com/koute/7391344) that follows your idea, it exposes the same compilation error for me (on Windows MSYS2), while it some other platform seems to work. MacOS probably?

Reply:
> None of them supports OpenGL2.0+ here by the required symbols. GLEW _does_ and links or sets up dispatches to their GL proc in some way when calling `glewInit()` at runtime.

You're correct. Maybe because Windows' `opengl32.lib/dll` is just implementing OpenGL 1.1 ?
So GLEW is still needed for Windows (at least) - I'll add it back.

> Look at this minimal example that follows your idea, it exposes the same compilation error for me (on Windows MSYS2), while it some other platform seems to work. MacOS probably?

I've tested on Linux only. The example linked works fine there (at least it compiles/links without errors).